src/PWAdditive.cpp

```diff
@@ -50,7 +55,11 @@ ddwaf_result PWAdditive::run(ddwaf_object newParameters, uint64_t timeLeft)
     // consistent across all possible timeout scenarios.
     if (timeLeft == 0)
     {
-        return returnErrorCode(DDWAF_ERR_TIMEOUT);
+        if (res) {
+            ddwaf_result &output = res.value();
+            output.timeout = true;
```

Comment:
If all you need is an out-parameter for whether a timeout occurred, it would make more sense to pass a pointer/reference to a bool only, rather than the full `ddwaf_result`.

Reply:
`ddwaf_result` is used below after the run.